### DISCUSSION TITLE
clock() question

### DISCUSSION THREAD

**Discussion by zjin-lcf** [intel/llvm#4443] [Q&A]
This example shows how to use the clock function to measure the performance of a kernel accurately.
https://github.com/zchee/cuda-sample/blob/master/0_Simple/clock/clock.cu

Is it useful for SYCL to have a clock() function ?

Thanks

**Comment by AlexeySachkov** [comment]
Hi @zjin-lcf, questions about the SYCL spec itself should be better asked in KhronosGroup/SYCL-Docs repo, where the spec source is published.

But anyway, regarding your question: there is an existing mechanism for measuring kernel performance: you can query [profiling information through events](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_event_information_and_profiling_descriptors), but you should create your command queue with [profiling enabled](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:queue-properties).

So, if you just want to time your kernels, that should be enough. If you want to precisely time _a part_ of your kernel, then apparently existing API won't be enough for you.

Tagging @gmlueck, @Pennycook and @mkinsner for awareness

**Reply by zjin-lcf** [reply]
Thanks for the suggestion.
